modules/contact/include/userobjects/WeightedGapUserObject.h

```diff
@@ -36,6 +36,53 @@ class WeightedGapUserObject : public MortarUserObject
    */
   virtual const ADVariableValue & contactPressure() const = 0;
 
+  /**
+   * @param node Node pointer
+   * @return The normal contact pressure at the node
+   */
+  virtual Real getNormalContactPressure(const Node * const /*node*/) const
+  {
+    mooseError("Not available in base class.");
+  };
+  /**
+   * @param node Node pointer
+   * @return The normal contact pressure at the node
+   */
+  virtual Real getNormalWeightedGap(const Node * const /*node*/) const;
+
+  /**
+   * @param node Node pointer
+   * @param component Component of the frictional pressure vector
+   * @return The frictional contact pressure at the node
+   */
+  virtual Real getFrictionalContactPressure(const Node * const /*node*/,
```

Comment:
> Maybe the aux kernel should be split up? 

@dschwen just mentioned that that's a possibility to tackle this issue. I initially wanted to avoid that, but since more code is being added for augmented Lagrange, I am fine with it.